serverless/middleware/auth.private.js

```diff
@@ -0,0 +1,38 @@
+/* global Twilio */
+'use strict';
+
+module.exports = async (context, event, callback) => {
+  const { PASSCODE, APP_EXPIRY, DOMAIN_NAME } = context;
+
+  const passcode = event.request.headers.authorization;
```

Comment:
@olipyskoty @timmydoza what do you think about adding user identity to the header also? So we don't have to separately add user identity as a parameter to requests? 

One way to do that would be using "basic" authorization which contains username and password. Here is an example of how this is encoded in swift:
```swift
    /// Returns a `Basic` `Authorization` header using the `username` and `password` provided.
    ///
    /// - Parameters:
    ///   - username: The username of the header.
    ///   - password: The password of the header.
    ///
    /// - Returns:    The header.
    public static func authorization(username: String, password: String) -> HTTPHeader {
        let credential = Data("\(username):\(password)".utf8).base64EncodedString()

        return authorization("Basic \(credential)")
    }
```
This would be easy on iOS since this networking library would do the encoding for me. I assume there is a JS package to encode and decode.

I guess only 3 out of the 6 endpoints use identity so this isn't really a must. But thought it was worth tossing around.

Reply:
I think we could, but it doesn't feel completely right. It seems odd to add the `user_identity` to the Authorization header when it's never going to be used for authentication or authorization. Since the API only uses the `user_identity` parameter to access specific resources in the Twilio API, I think it makes sense that `user_identity` should be in the request body and not the Authorization header. 

Also, I fear that adding Basic Auth might give the impression that this app is more secure than it really is. I wouldn't want to give the impression that one user can't change another user's hand raised status (for example). The passcode is only intended to protect the endpoints from unauthorized use, not to protect each individual user's resources from other malicious users. 

Down the road, I think we could implement more robust authentication that protects users from other users using Bearer Authentication. The Twilio access token that is given to every user can be used for this. The backend is able to tell that it is signed correctly (the BE has access to the right secret for this), the user's identity is embedded in the token, and each user only has access to their token.

Reply:
Very good points thanks!